### DISCUSSION TITLE
Csla.DataPortalException at serialisation of object list

### DISCUSSION THREAD

**Discussion by uwefms** [MarimerLLC/csla#1993] [General]
Hi everybody,

is there a way to serialize a list of string pair objects (used in a combobox) in CSLA wasm. This is working fine in serverside blazor but in wasm I get a notsupported error:

Csla.DataPortalException: Die Serialisation von Objekt-Type wird nicht unterstützt. (Serialisation of Objekt-Type is not supported)(System.Collections.Generic.List`1[[Dual_Test.Dal.AppMandant, Dual_Test.Dal, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]])

I used to work in server side blazor only. Now I try to run a dual mode app (based on ProjectTracker) and I noticed the problem in the CSB part. I am using CSLA 5.4.0 in .net5. Workaround could be to convert the string pair object to one string and reconvert it when needed.

   regards

       Uwe

**Comment by rockfordlhotka** [comment]
CSLA increasingly relies on `MobileFormatter` to serialize object graphs. Microsoft seems to be slowly getting rid of `BinaryFormatter`, so this is working out well.

However, `MobileFormatter` has some restrictions on what it can serialize. You need to use the types in `Csla.Core`, such as `MobileList`.

**Reply by uwefms** [reply]
Hi Rocky,

  thanks for answering. I used `MobileList<AppMandant>`, which caused the error. `MobileList<string>` was fine.

So trying to change AppMandant from a "plain" class to a csla businessobject, should do the trick ?

   regards

      Uwe

**Reply by rockfordlhotka** [reply]
Everything in the object graph needs to implement `IMobileObject` or be a primitive type.

`Csla.Core` includes several base types that implement `IMobileObject` on your behalf, so if you want to start from scratch you can use them (or implement the interface yourself). Or you can subclass the higher level types like `BusinessBase` if you want the rules engine and other features it provides.

**Reply by uwefms** [reply]
Hi Rocky,

   thank you very much for taking your time and pionting me into the right direction. Everything is working as expected.

   regards

       Uwe
